autotest/gcore/cog.py

```diff
@@ -586,3 +586,59 @@ def test_cog_invalidation_by_metadata_change():
 
     with gdaltest.error_handler():
         gdal.GetDriverByName('GTiff').Delete(filename)
+
+
+###############################################################################
+# Test a tiling scheme with a CRS with northing/easting axis order
+# and non power-of-two ratios of scales.
+
+
+def test_cog_northing_easting_and_non_power_of_two_ratios():
+
+    filename = '/vsimem/cog.tif'
+    ds = gdal.Translate(filename, 'data/byte.tif',
+        options='-of COG -a_srs EPSG:2193 -a_ullr 1000001 5000001 1000006.6 4999995.4 -co TILING_SCHEME=NZTM2000 -co ALIGNED_LEVELS=2')
+    assert ds.RasterXSize == 1280
+    assert ds.RasterYSize == 1280
+    b = ds.GetRasterBand(1)
+    assert [(b.GetOverview(i).XSize, b.GetOverview(i).YSize) for i in range(b.GetOverviewCount())] == [(512, 512), (256, 256)]
+
+    gt = ds.GetGeoTransform()
+    # 0.28 m/pixel corresponds to OGC ScaleDenom = 1000 / zoom level=14 in NZTM2000
+    assert gt == pytest.approx((999872, 0.28, 0, 5000320, 0, -0.28), abs=1e-8)
+
+    # Check that gt origin matches the corner of a tile at zoom 14
+    x0_NZTM2000 = -1000000
+    y0_NZTM2000 = 10000000
+    res = gt[1]
+    tile_x = (gt[0] - x0_NZTM2000) / (256 * res)
+    assert tile_x == pytest.approx(int(tile_x + 0.5))
+    tile_y = (y0_NZTM2000 - gt[3]) / (256 * res)
+    assert tile_y == pytest.approx(int(tile_y + 0.5))
+
+    # Check that overview dimension of 512 corresponds to the resolution of zoom level=13 / OGC ScaleDenom = 2500
+    assert 512. / 1280 == 1000. / 2500.
```

Comment:
Rather than use magic numbers, can we use named variables here? e.g rastersize, oversize block size

Reply:
done

Reply:
Thanks!